File: C++/Kyu8/Sort and Star.cpp
/*
|_______________________________________________________________________________
| @Solver         : GziXnine
| @Link           : https://www.codewars.com/kata/57cfdf34902f6ba3d300001e
| @description    : You will be given a list of strings. You must sort it
|                   alphabetically (case-sensitive, and based on the ASCII
|                   values of the chars) and then return the first value.
|
|                   The returned value must be a string and have "***" between
|                   each of its letters.
|
|                   You should not remove or add elements from/to the array.
|_______________________________________________________________________________
*/
#include <vector>
#include <string>
#include <algorithm>

std::string twoSort(std::vector<std::string> s)
{
  sort(s.begin(), s.end());
  std::string newS = s[0];
  for (int i = 1; i < newS.size(); i += 4)
    newS.insert(i, "***");
  return newS;
}

File: JavaScript/Kyu8/Sort and Star.js
/*
|_______________________________________________________________________________
| @Solver         : GziXnine.
| @Link           : https://www.codewars.com/kata/57cfdf34902f6ba3d300001e
| @description    : Given a list of strings, sort it alphabetically (case-sensitive)
|                   based on the ASCII values of the characters, and return the first
|                   value in the sorted list. The returned value should have "***" 
|                   inserted between each of its letters.
|                   Example:
|                   Input: ["bitcoin", "apple", "banana"]
|                   Output: "a***p***p***l***e"
|                   Notes: The sorting is case-sensitive and the first value is returned
|                   as a string with "***" between each character.
|_______________________________________________________________________________
*/
function twoSort(s) {
  return s.sort()[0].split("").join("***");
}
